lib/Runtime/Library/WebAssemblyModule.cpp

```diff
@@ -37,12 +36,13 @@ WebAssemblyModule::WebAssemblyModule(Js::ScriptContext* scriptContext, const byt
     m_binaryBufferLength(binaryBufferLength),
     m_customSections(nullptr)
 {
+    m_alloc = HeapNew(ArenaAllocator, _u("WebAssemblyModule"), scriptContext->GetThreadContext()->GetPageAllocator(), Js::Throw::OutOfMemory);
```

Comment:
This is because adding
```cpp
protected:  
    DEFINE_VTABLE_CTOR(WebAssemblyModule, DynamicObject); 
    DEFINE_MARSHAL_OBJECT_TO_SCRIPT_CONTEXT(WebAssemblyModule);
```

Creates a constructor with `VirtualTableInfoCtorEnum` as argument. This means, I need a default constructor for the ArenaAllocator. I tried initially to add a such a constructor to ArenaAllocator, but it proved to be more difficult than I expected. 
I then decided to switch to a allocate the ArenaAllocator on the heap and manage the memory using Finalize.

I could try/catch the other allocators and free the m_alloc memory if an exception occurs. But the only kind of exceptions we can encounter are OOM exception, I doubt it is worth protecting for this case.